Source/bmalloc/libpas/src/libpas/pas_deallocate.c

```diff
@@ -64,6 +65,33 @@ bool pas_try_deallocate_known_large(void* ptr,
     return true;
 }
 
+bool pas_try_deallocate_pgm_large(void* ptr,
+                                  const pas_heap_config* config_ptr)
+{
+    uintptr_t begin = (uintptr_t)ptr;
+    pas_heap_lock_lock();
+    if (config_ptr->pgm_enabled && pas_probabilistic_guard_malloc_check_exists(begin)) {
+        pas_probabilistic_guard_malloc_deallocate((void *) begin);
+        pas_heap_lock_unlock();
+        return true;
+    }
+    pas_heap_lock_unlock();
+    return false;
+}
+
+bool pas_check_pgm_entry_exists(void *ptr)
+{
+    uintptr_t begin = (uintptr_t)ptr;
+    pas_heap_lock_lock();
+    bool check = pas_probabilistic_guard_malloc_check_exists(begin);
+    if (check) {
+        pas_heap_lock_unlock();
+        return true;
+    }
+    pas_heap_lock_unlock();
+    return false;
```

Comment:
This could be written a bit simpler (without a branch):

```suggestion
    pas_heap_lock_lock();
    bool result = pas_probabilistic_guard_malloc_check_exists(begin);
    pas_heap_lock_unlock();
    return result;
```

Reply:
Fixed